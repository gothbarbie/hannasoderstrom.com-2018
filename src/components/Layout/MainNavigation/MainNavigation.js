import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'

import Icon from '../../Icon'

const Nav = styled.nav`
  display: flex;
  align-items: center;

  & ul {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  & li {
    margin: 0 1rem;
  }

  & a {
    & :hover {
      border-bottom: 4px solid white;
    }
  }
`

const HomeLinkWithActive = styled(Link)`
  color: ${({ active }) =>
    active === 'active' ? '#fff' : 'rgba(255, 255, 255, 0.65)'};
  text-decoration: none;
  border-bottom: ${({ active }) =>
    active === 'active' ? '4px solid white' : ''};
`

const LinkWithActive = HomeLinkWithActive.extend`
  padding-bottom: 0.375rem;
`

const MainNavigation = () => (
  <Nav>
    <ul>
      <li>
        <HomeLinkWithActive
          active={window.location.pathname === '/' ? 'active' : ''}
          to="/"
        >
          <Icon name="home" />
        </HomeLinkWithActive>
      </li>
      <li>
        <LinkWithActive
          active={window.location.pathname === '/recently' ? 'active' : ''}
          to="/recently"
        >
          Recently
        </LinkWithActive>
      </li>
      <li>
        <LinkWithActive
          active={window.location.pathname === '/experience' ? 'active' : ''}
          to="/experience"
        >
          Experience
        </LinkWithActive>
      </li>
      <li>
        <LinkWithActive
          active={window.location.pathname === '/contact' ? 'active' : ''}
          to="/contact"
        >
          Contact
        </LinkWithActive>
      </li>
    </ul>
  </Nav>
)

export default MainNavigation
