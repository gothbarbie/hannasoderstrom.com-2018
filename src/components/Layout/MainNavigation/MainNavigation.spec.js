import React from 'react'
import { shallow } from 'enzyme'

import MainNavigation from './index'

describe('components/MainNavigation', () => {
  let component

  beforeEach(() => {
    component = shallow(<MainNavigation />)
  })

  it('renders', () => {
    expect(component).toMatchSnapshot()
  })
})
