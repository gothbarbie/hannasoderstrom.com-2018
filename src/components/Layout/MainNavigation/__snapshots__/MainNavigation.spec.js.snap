// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MainNavigation renders 1`] = `
<styled.nav>
  <ul>
    <li>
      <Styled(Link)
        active="active"
        data-test="home-link"
        to="/"
      >
        <Icon
          name="home"
        />
      </Styled(Link)>
    </li>
    <li>
      <Styled(Link)
        active=""
        data-test="recently-link"
        to="/recently"
      >
        Recently
      </Styled(Link)>
    </li>
    <li>
      <Styled(Link)
        active=""
        data-test="experience-link"
        to="/experience"
      >
        Experience
      </Styled(Link)>
    </li>
    <li>
      <Styled(Link)
        active=""
        data-test="contact-link"
        to="/contact"
      >
        Contact
      </Styled(Link)>
    </li>
  </ul>
</styled.nav>
`;
